client/src/pages/Join/Join.jsx

```diff
@@ -1,37 +1,61 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
 import io from 'socket.io-client';
+import PropTypes from 'prop-types';
 
 import './Join.css';
 
-import UserContext from '../../context/UserContext';
+import { UserContext } from '../../context/UserContext';
 import youthIlustration from '../../assets/friends.svg';
 import shape from '../../assets/shape-1.svg';
+import Modal from '../../components/Modal/Modal';
 
-// <a href='https://www.freepik.com/vectors/social-media'>Social media vector created by stories - www.freepik.com</a> 
-// abstract <a href='https://www.freepik.com/vectors/banner'>Banner vector created by freepik - www.freepik.com</a>
-
-const Join = () => {
+const Join = ({ history }) => {
   let socket;
   const ENDPOINT = process.env.NODE_ENV === 'production' ? window.location.origin : 'localhost:5000';
 
   const [formName, setFormUsername] = React.useState('');
   const [formRoom, setFormRoom] = React.useState('');
+  const [error, setError] = React.useState('');
 
-  const context = React.useContext(UserContext);
+  const {
+    showModal,
+    setShowModal,
+    setSocket,
+    setName,
+    setRoom,
+  } = React.useContext(UserContext);
 
   const join = (event) => {
     if (!formName || !formRoom) {
       event.preventDefault();
       return;
     }
+
     socket = io(ENDPOINT, {
       reconnection: false,
     });
-    socket.emit('join', { name: formName, room: formRoom }, (error) => { if (error) console.log(error); });
-    context.setSocket(socket);
-    context.setName(formName);
-    context.setRoom(formRoom);
+
+    socket.emit('join', { name: formName, room: formRoom }, function(error) { 
+      if (error) {
+      setError(error);
+      return;
+      }
+    });
+
+    socket.on('connect_error', () => {
+      setShowModal(true);
+    });
+
+    socket.on('error', () => {
+      setShowModal(true);
+    });
+
+    socket.on('login_successful', () => {
+      setSocket(socket);
+      setName(formName);
+      setRoom(formRoom);
+      history.push('/chat');
```

Comment:
Consider using the [useHistory](https://reactrouter.com/web/api/Hooks/usehistory) hook from `react-router-dom`.